include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1567,7 +1568,8 @@ MatrixFree<dim, Number, VectorizedArrayType>::initialize_indices(
                 if (cell < n_cell_batches())
                   {
                     // locally-owned cell
-                    cells_locally_owned.emplace_back(index, global_cell_index);
```

Comment:
The `index` was wrong. I did not intend to use the matrix-free cell index but the "dealii" index.